### PR TITLE
Adapt to Python3.6+ and Matplotlib 3.8+

### PR DESCRIPTION
**Pull request by PiotrZakrzewski**
- When running on python 3.6+ dnaplotlib fails on `open("filename", "rU")`, the use of `U` is redundant in Python3 and since Python 3.6 is officially deprecated, on the currently support versions it leads to an error.
- Minor fixes to some gallery scripts (after changing dnaplotlib I ran the examples to see if they still work
- Upgrade to Matplotlib 3.8 along with minor changes

Tested on Python 3.11.4 by running the galery scripts

### CODE REVIEW

File: gallery/repressilator_animate/repressilator_figure.py [hunk 1]
```diff
@@ -128,7 +128,7 @@ def movie(ts, ymtet, ymlac, ymgamma, ytet, ylac, ygamma):
 		plt.plot(ts[int(t*10)], ygamma[int(t*10)], '.', color=[0.38, 0.65, 0.87], markersize=6.0)
 		ax = plt.subplot(gs[2])
 		plot_construct(ax, t, ymtet, ymlac, ymgamma, ytet, ylac, ygamma)
-		plt.savefig("movie/repressilator_t{}.jpg".format(t), dpi=300)
```
Comment (PiotrZakrzewski) [inline gallery/repressilator_animate/repressilator_figure.py:131]: The script fails on this dir, it does not exist and is not created.

**Comment by PiotrZakrzewski** [comment]
I have just realized that `sbol`  (python lib for version1 of the standard) lib not available for Python3 at all, so both of the sbol gallery examples do not work on python3. Lemme see how hard it is to adapt it. 
@chofski  which SBOL version do you aim to support - v2 or v3?

File: dnaplotlib/sbol/sbolplotlib.py [hunk 2]
```diff
@@ -15,54 +15,44 @@
...
-        'SO_0005850': 'PrimerBindingSite',
-        'SO_0001694': '5StickyRestrictionSite',
-        'SO_0001690': '3StickyRestrictionSite',
-        'SO_0000001': 'UserDefined',
-        'SO_0001978': 'Signature',
+            "SO:0000167": "Promoter",
+            "SO:0000316": "CDS",
+            "SO:0000141": "Terminator",
+            "SO:0000552": "RBS",
+            "SO:0000139": "RBS",
```
Comment (PiotrZakrzewski) [inline dnaplotlib/sbol/sbolplotlib.py:35]: I added this symbol http://www.sequenceontology.org/browser/current_release/term/SO:0000139
used the same label as it appears to be the parent of SO:0000552 in the hierarchy.

File: gallery/sbol_visual/gene_cassette.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import sbol3 as sbol
```
Comment (PiotrZakrzewski) [inline gallery/sbol_visual/gene_cassette.py:1]: I added this script to (re)generate the SBOLv3 xml for the Gene Cassete example design

File: gallery/sbol_visual/sbol_visual.py [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
 doc = sbol.Document()
-doc.read('gene_cassette.sbol')
+doc.read("gene_cassette.xml")
 
-# In this case, we know ahead of time the URI of the design.  In some cases, you may have to explore the doc's components to find the design you are looking for
-design = doc.components['http://sbolstandard.org/examples/Design']
-print(design.display_id)
+# In this case, we know ahead of time the name of the design. In some cases, you may have to explore the doc's components to find the design you are looking for
+design = doc.find("GeneCassette")
+print(design.name)
```
Comment (PiotrZakrzewski) [inline gallery/sbol_visual/sbol_visual.py:14]: according to the SBOLv3.1 standard it is the name that is used for display purposes
see https://sbolstandard.org/docs/SBOL3.1.0.pdf
page 15

**Comment by PiotrZakrzewski** [comment]
@chofski I added the necessary adaptations to make sbol_visual example to work (updated to work with sbolv3 standard and pysbol3 lib). Some remarks:
- the original Gene Cassete example skipped one of the subcomponents due to SO number mismatch, now the visualization looks different, as it shows all four.
- I did not adapt the sbol_viewer gallery example, this one is MUCH bigger, a material for a separate PR.

**Comment by chofski** [comment]
All looks good. Thanks for the updates.

**Pull request merged**